test/suites/clustering.sh

```diff
@@ -3543,74 +3543,97 @@ test_clustering_evacuation() {
+
   # For debugging
   LXD_DIR="${LXD_TWO_DIR}" lxc list
 
+  # Move c7 to node1 to test "volatile.cluster.target" handling post-evacuation.
+  LXD_DIR="${LXD_TWO_DIR}" lxc stop c7 -f
+  LXD_DIR="${LXD_TWO_DIR}" lxc move c7 --target=node1
+  LXD_DIR="${LXD_TWO_DIR}" lxc start c7
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c7 | grep -cF "Location: node1")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c7 | grep -cF "Status: RUNNING")" = 1 ]
+
   # Evacuate first node
   LXD_DIR="${LXD_TWO_DIR}" lxc cluster evacuate node1 --force
 
   # Ensure the node is evacuated
-  LXD_DIR="${LXD_TWO_DIR}" lxc cluster show node1 | grep -F "status: Evacuated"
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc cluster show node1 | grep -cF "status: Evacuated")" = 1 ]
 
   # For debugging
   LXD_DIR="${LXD_TWO_DIR}" lxc list
 
   # Check instance status
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c1 | grep -xF "Status: RUNNING"
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc info c1 | grep -xF "Location: node1" || false
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c2 | grep -xF "Status: RUNNING"
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc info c2 | grep -xF "Location: node1" || false
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c3 | grep -xF "Status: STOPPED"
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c3 | grep -xF "Location: node1"
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c4 | grep -xF "Status: RUNNING"
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc info c4 | grep -xF "Location: node1" || false
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c5 | grep -xF "Status: STOPPED"
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc info c5 | grep -xF "Location: node1" || false
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c6 | grep -xF "Status: RUNNING"
-  LXD_DIR="${LXD_TWO_DIR}" lxc info c6 | grep -xF "Location: node2"
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c1 | grep -cF "Status: RUNNING")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c1 | grep -cF "Location: node1")" = 0 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c2 | grep -cF "Status: RUNNING")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c2 | grep -cF "Location: node1")" = 0 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c3 | grep -cF "Status: STOPPED")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c3 | grep -cF "Location: node1")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c4 | grep -cF "Status: RUNNING")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c4 | grep -cF "Location: node1")" = 0 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c5 | grep -cF "Status: STOPPED")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c5 | grep -cF "Location: node1")" = 0 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c6 | grep -cF "Status: RUNNING")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c6 | grep -cF "Location: node2")" = 1 ]
+
+  # Check "volatile.target.cluster" (initial target cluster group) is respected.
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c7 | grep -cF "Status: RUNNING")" = 1 ]
+  [ "$(LXD_DIR="${LXD_TWO_DIR}" lxc info c7 | grep -cF "Location: node3")" = 1 ]
```

Comment:
This was reworked for fewer `lxc` calls in `main` but I am now wondering if we shouldn't go with a single `lxc list -f json > list.json` followed by a bunch of `jq -e 'assertion checks for each instance' < list.json`.